src/transactions/isWorkable.ts

```diff
@@ -0,0 +1,26 @@
+import type { Contract } from 'ethers';
+
+/**
+ * @notice Helper function to check if is workable
+ *
+ * @param job				An instance of the job we wish to call.
+ * @param methodArguments	The arguments of the work function we wish to call.
+ *
+ * @return True if is workable.
+ */
+export async function checkIsWorkable(job: Contract, methodArguments: Array<number | string>): Promise<boolean> {
+	try {
+		await job.callStatic.work(...methodArguments);
+		return true;
+	} catch (error: unknown) {
+		if (
+			error instanceof Error &&
+			!error.message.includes('NotWorkable()') &&
+			!error.message.includes('V2Keep3rJob::work:not-workable')
+		) {
+			console.log(`Failed when attempting to call work statically. Message: ${error.message}. Returning.`);
+		}
+
+		return false;
```

Comment:
hmhmmmm i wouldn't leave this here, the `callStatic.work` is defined by each job owner so is hard to abstract.

i would accept in such case a method that would make the static call and return the output, but don't mix with the `isWorkable` check, a JS mehtod that needs to at every block check for `workable`, or maybe compare nonces and stuff, don't import it from here 🙏